### PR TITLE
chore(deps): upgrade @nestjs/axios to 3.0.1

### PR DESCRIPTION
**Pull request by axtx4869**
update @nestjs/axios and other packages with incorrect peer dependency

----

fix https://github.com/OpenAPITools/openapi-generator-cli/issues/719

### CODE REVIEW

File: package.json
Comment (wvanderdeijl) [inline package.json:111]: I wouldn't recommended updating nx directly in the package.json. Shouldn't we use `nx migrate` (https://nx.dev/core-features/automate-updating-dependencies) as that usually does much more work than just updating dependencies.

Comment (axtx4869) [reply package.json]: I'm sorry, I should have read nx's document and shouldn't include nx update in this PR.
I only updated something related to the issue

File: package.json
Comment (alexan) [inline package.json:89]: why do you set a specific `axios` version here. `@nestjs/axios` is using `1.6.2` issn't this just make this harder to maintain?

Comment (axtx4869) [reply package.json:89]: If I remove axios, an warning occurs like below.
> warning " > @nestjs/axios@3.0.1" has unmet peer dependency "axios@^1.3.1"

Is it okay to ignore this, or should I add axios to devDependencies?

Comment (alexan) [reply package.json:89]: as this is a library I would say this warning is okay as the consuming package should install its axios version

Comment (axtx4869) [reply package.json]: I see, I removed `axios` from dependencies

**Review by alexan — APPROVED** [review]

**Comment by manan19** [comment]
Thanks for putting this together. Will it be released after this PR is merged? context https://github.com/advisories/GHSA-wf5p-g6vw-rhxx

File: package.json
Comment (axtx4869) [reply package.json]: @wvanderdeijl 
Could you re-review this PR?
I need approval from a maintainer to run ci workflow.

**Comment by wing328** [comment]
CI failed: https://github.com/OpenAPITools/openapi-generator-cli/actions/runs/6955846991/job/19127935712?pr=722

can you please take a look when you've time?

thanks again for the PR.

**Comment by wing328** [comment]
still some failed tests: https://github.com/OpenAPITools/openapi-generator-cli/actions/runs/7077821666/job/19262635518?pr=722

please take a look when you've time

**Comment by axtx4869** [comment]
@wing328 
Could you rerun workflow? I am not sure it will work well
thank you for your help

**Comment by wing328** [comment]
Just reran the workflow. Let's see how that goes

**Comment by wing328** [comment]
unfortunately the tests still failed:
```
Done in 9.16s.

> examples@1.0.0 oa
> openapi-generator-cli version

node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module 'axios'
Require stack:
- /home/runner/work/openapi-generator-cli/openapi-generator-cli/examples/node_modules/@nestjs/axios/dist/http.module.js
- /home/runner/work/openapi-generator-cli/openapi-generator-cli/examples/node_modules/@nestjs/axios/dist/index.js
- /home/runner/work/openapi-generator-cli/openapi-generator-cli/examples/node_modules/@nestjs/axios/index.js
- /home/runner/work/openapi-generator-cli/openapi-generator-cli/examples/node_modules/@openapitools/openapi-generator-cli/main.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1[143](https://github.com/OpenAPITools/openapi-generator-cli/actions/runs/7102942856/job/19409541420?pr=722#step:5:144):19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at Object.<anonymous> (/home/runner/work/openapi-generator-cli/openapi-generator-cli/examples/node_modules/@nestjs/axios/dist/http.module.js:16:33)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/runner/work/openapi-generator-cli/openapi-generator-cli/examples/node_modules/@nestjs/axios/dist/http.module.js',
    '/home/runner/work/openapi-generator-cli/openapi-generator-cli/examples/node_modules/@nestjs/axios/dist/index.js',
    '/home/runner/work/openapi-generator-cli/openapi-generator-cli/examples/node_modules/@nestjs/axios/index.js',
    '/home/runner/work/openapi-generator-cli/openapi-generator-cli/examples/node_modules/@openapitools/openapi-generator-cli/main.js'
```

**Comment by axtx4869** [comment]
@wing328 Sorry for taking a while to fix ci failure.
I would like you to rerun the workflow

**Comment by axtx4869** [comment]
@wing328 
Could you run workflow again🙏
I found a warning which may cause ci error and changed axios version specification

> warning "/home/runner/work/openapi-generator-cli/openapi-generator-cli/package.tgz > @nestjs/axios@3.0.1" has unmet peer dependency "axios@^1.3.1".

**Comment by wing328** [comment]
done but the tests still failed with the same error message

**Comment by axtx4869** [comment]
@wing328 
I changed jest config like https://github.com/OpenAPITools/openapi-generator-cli/issues/719#issuecomment-1865856783
I wanna see how the workflow goes

**Comment by wing328** [comment]
done but got the same error

**Comment by simonhammes** [comment]
I don't know a lot about NX, but this seems like a bug in NX related to the implicit dependency functionality.

`axios` was specified as a direct dependency as part of this PR, but it does not get added to the generated `package.json`, which causes the `MODULE_NOT_FOUND` errors in CI.

See https://github.com/OpenAPITools/openapi-generator-cli/actions/runs/7287055215/job/19881267299?pr=722#step:5:52

Maybe upgrading NX first would help? :shrug: 

Or is there a way to specify an _explicit dependency_ with NX?

**Comment by wing328** [comment]
@simonhammes thanks for the suggestion

@axtx4869 can you please also update NX as part of this PR?

(I saw the comment from @wvanderdeijl before)

**Comment by elmarbeckmann** [comment]
Any updates on this issue? I see that NX (17.2.x) now has axios@1.6.5 as a dependency which fixes the vulnerability there.

**Comment by wing328** [comment]
@elmarbeckmann do you mind filing a PR on top of this change with the NX upgraded to newer version?

**Comment by elmarbeckmann** [comment]
> @elmarbeckmann do you mind filing a PR on top of this change with the NX upgraded to newer version?

here you go https://github.com/OpenAPITools/openapi-generator-cli/pull/722

**Comment by wing328** [comment]
closed via https://github.com/OpenAPITools/openapi-generator-cli/pull/730

**Pull request closed without merging**
